### PR TITLE
Add topic statistics helper plot

### PR DESCRIPTION
**Pull request by carlossvg**
This PR adds a script to plot metrics from the topic statistics topic.

- [x] Test script with all the options
 
## Instructions

1. Enable `enable_topic_stats: True` param

2. Launch demo

```bash
 ros2 launch pendulum_bringup pendulum_bringup.launch.py
```
3. Log controller_stats in csv format

```bash
ros2 topic echo /controller_stats --csv > controller_stats.csv
```
4. Run script passing the csv file:

```bash
python3 topic_statistics_plot.py path/to/controller_stats.csv --show -m message_period
```

5. Check plot

Figure_1

```bash
python3 topic_statistics_plot.py path/to/controller_stats.csv --show -m message_age
```
Figure_2

## Notes

- Apparently the topic statistics maximum and minimum are not calculated for the measurement window but for the whole time. This makes the plot less informative.
- The script assumes the order of the metrics is fixed. This could change in the future. This can be improved by using the data_types information.
- The script will fail if there are NaN.

### CODE REVIEW

File: pendulum_analysis/topic_statistics_analysis/topic_statistics_plot.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    filename = args.filename
+    topicname = args.topicname
+    show = args.show
+    metric = args.metric
+    outfile = filename + '_plot'
+    if args.outfile is not None:
+        outfile = args.outfile
+    else:
+        if topicname:
+            outfile = topicname + '_' + metric
```
Comment (LanderU) [inline pendulum_analysis/topic_statistics_analysis/topic_statistics_plot.py]:
```suggestion
            outfile = f"{topicname}_{metric}"
```

File: pendulum_analysis/topic_statistics_analysis/topic_statistics_plot.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        else:
+            outfile = 'topic_statistics' + '_' + metric
```
Comment (LanderU) [inline pendulum_analysis/topic_statistics_analysis/topic_statistics_plot.py]:
```suggestion
            outfile = f"{topic_statistics}_{metric}"
```

File: pendulum_analysis/topic_statistics_analysis/topic_statistics_plot.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+    # Plot topic statistic metrics
+    plt.figure()
+    ax = plt.gca()
+    df.plot(kind='line', x='time', y='average', ax=ax)
+    df.plot(kind='line', x='time', y='minimum', ax=ax)
+    df.plot(kind='line', x='time', y='maximum', ax=ax)
+
+    ax.set_xlabel("Time (s)")
+    ax.set_ylabel(metric + " (" + unit + ")")
```
Comment (LanderU) [inline pendulum_analysis/topic_statistics_analysis/topic_statistics_plot.py]:
```suggestion
    ax.set_ylabel(f"{metric} ({unit} )")
```

File: pendulum_analysis/topic_statistics_analysis/topic_statistics_plot.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    plt.title(metric + ' vs. time')
```
Comment (LanderU) [inline pendulum_analysis/topic_statistics_analysis/topic_statistics_plot.py]:
```suggestion
    plt.title(f"{metric}  vs. time")
```

File: pendulum_analysis/topic_statistics_analysis/topic_statistics_plot.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    if show:
+        plt.show()
+    plt.savefig(outfile + '.svg')
```
Comment (LanderU) [inline pendulum_analysis/topic_statistics_analysis/topic_statistics_plot.py]:
```suggestion
    plt.savefig(f"{outfile}.svg")
```

**Review by LanderU — COMMENTED** [review]
Some minor python style suggestions added.

File: pendulum_analysis/topic_statistics_analysis/topic_statistics_plot.py (hunk 1 above)
Comment (carlossvg) [reply pendulum_analysis/topic_statistics_analysis/topic_statistics_plot.py]: Updated

File: pendulum_analysis/topic_statistics_analysis/topic_statistics_plot.py (hunk 2 above)
Comment (carlossvg) [reply pendulum_analysis/topic_statistics_analysis/topic_statistics_plot.py]: This should be: `outfile = f"topic_statistics_{metric}"`

File: pendulum_analysis/topic_statistics_analysis/topic_statistics_plot.py (hunk 3 above)
Comment (carlossvg) [reply pendulum_analysis/topic_statistics_analysis/topic_statistics_plot.py]: Updated

File: pendulum_analysis/topic_statistics_analysis/topic_statistics_plot.py (hunk 4 above)
Comment (carlossvg) [reply pendulum_analysis/topic_statistics_analysis/topic_statistics_plot.py]: Updated

File: pendulum_analysis/topic_statistics_analysis/topic_statistics_plot.py (hunk 5 above)
Comment (carlossvg) [reply pendulum_analysis/topic_statistics_analysis/topic_statistics_plot.py]: Updated

**Review by LanderU — APPROVED** [review]
LGTM!

**Pull request merged**